python/lsst/meas/base/forcedPhotCoadd.py

```diff
@@ -87,6 +93,11 @@ class ForcedPhotCoaddConnections(pipeBase.PipelineTaskConnections,
         dimensions=["band", "skymap", "tract", "patch"],
     )
 
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+        if config.footprintDatasetName != "scarletDeblendModels":
+            self.inputs -= set(("scarletModels",))
```

Comment:
```suggestion
            self.inputs.remove("scarletModels")
```

Reply:
I didn't realize that it could be done this way. I was just copying https://github.com/lsst/pipe_tasks/blob/main/python/lsst/pipe/tasks/multiBand.py#L682-L705 . I'll fix those in `pipe_tasks` too.